pkg/traversal/traversal.go

```diff
@@ -57,27 +57,22 @@ func (s *service) Traverse(ctx context.Context, addr swarm.Address, iterFn swarm
 	}
 
 	ls := loadsave.NewReadonly(s.store)
-	switch mf, err := manifest.NewDefaultManifestReference(addr, ls); {
-	case errors.Is(err, manifest.ErrInvalidManifestType):
-		break
-	case err != nil:
-		return fmt.Errorf("traversal: unable to create manifest reference for %q: %w", addr, err)
+	mf, _ := manifest.NewDefaultManifestReference(addr, ls)
```

Comment:
I am not sure if it is safe to remove handling of ErrInvalidManifestType error.

Reply:
So currently this function cannot return an error. The New function only returns an error if the type is incorrect and `DefaultManifestReference` gives the Mantaray type. Also the `NewXXXReference` call doesn't do any get till other operations are done.

Reply:
I would even remove error return value from NewDefaultManifestReference to make that clear. But that is probably for a different PR.